### PR TITLE
meta pkg: wrap for traceability

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

When your meta pkg is failing, it helps to understand what went wrong
image

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @tchaton @rohitgr7 @borda

### CODE REVIEW

File: .actions/setup_tools.py [hunk 1]
```diff
@@ -270,6 +270,37 @@ def prune_comments_docstrings(lines: List[str]) -> List[str]:
...
+    if not_empty == 0:
+        return body
+    body = ["try:"] + [f"    {ln}" if ln else "" for ln in body]
+    body += [
+        "",
+        "except ImportError as err:",
+        "",
+        "    from os import linesep",
+        f"    from {pkg} import __version__",
+        f"    msg = f'Your meta package was build for `{pkg}=={ver}`," + " but you are running {__version__}'",
```
Comment (awaelchli) [inline .actions/setup_tools.py:285]:
```suggestion
        f"    msg = f'Your meta package was built for `{pkg}=={ver}`," + " but you are running {__version__}'",
```

Comment (awaelchli) [reply .actions/setup_tools.py:285]: I don't think a user knows what a meta package is (frankly, I don't even feel like I do). 
Isn't this just `lightning`?

File: .actions/setup_tools.py [hunk 2]
```diff
@@ -270,6 +270,37 @@ def prune_comments_docstrings(lines: List[str]) -> List[str]:
     return body
 
 
+def wrap_try_except(body: List[str], pkg: str, ver: str) -> List[str]:
+    """Wrap the file with try/except for better traceability of imp0rt misalignment."""
```
Comment (awaelchli) [inline .actions/setup_tools.py:274]:
```suggestion
    """Wrap the file with try/except for better traceability of import misalignment."""
```

**Review by awaelchli — APPROVED** [review]

File: .actions/setup_tools.py (hunk 1 above)
Comment (Borda) [reply .actions/setup_tools.py:285]: good point

**Review by manskx — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
Neat !

**Pull request merged**